internal/services/authorization/validate/role_resource_id.go

```diff
@@ -0,0 +1,22 @@
+package validate
+
+import (
+	"fmt"
+	"strings"
+)
+
+func RoleResourceID(input interface{}, key string) (warnings []string, errors []error) {
+	v, ok := input.(string)
```

Comment:
Shall we use the `parse.RoleDefinitionId` here to validate whether the input is parsable?

Reply:
I am afraid not, because the input of [parse.RoleDefinitionId](https://github.com/hashicorp/terraform-provider-azurerm/blob/99e02144c7fe6130132907ec3ffeafb6d7bb57d4/internal/services/authorization/parse/role_definition.go#L17) needs to be a combination of `resourceId|scope`, but the `role_definition_id` used in this resouce is a `resouceId` only.